package NCommunism.NProto;

import "tools/clustermaster/communism/core/definition.proto";

enum EMessageType {
    MT_HEARTBEAT = 0;
    MT_DEFINE = 1;
    MT_UNDEF = 2;
    MT_REQUEST = 3;
    MT_CLAIM = 4;
    MT_DISCLAIM = 5;
    MT_DETAILS = 6;
    MT_GRANT = 7;
    MT_REJECT = 8;
    MT_GROUP = 9;
}

message THeartbeatMessage {
}

message TDefineMessage {
    required uint32 ActionId = 1;
    required uint32 Key = 2;
    optional TDefinition Definition = 3;
    optional uint32 Origin = 4;
}

message TUndefMessage {
    required uint32 Key = 2;
}

message TRequestMessage {
    required uint32 ActionId = 1;
    required uint32 Key = 2;
}

message TClaimMessage {
    required uint32 ActionId = 1;
    required uint32 Key = 2;
}

message TDisclaimMessage {
    required uint32 Key = 2;
}

message TDetailsMessage {
    required uint32 Key = 1;
    required TDetails Details = 2;
}

message TGrantMessage {
    required uint32 ActionId = 1;
    required uint32 Key = 2;
}

message TRejectMessage {
    required uint32 ActionId = 1;
    required uint32 Key = 2;
    optional string Message = 3;
}

message TGroupMessage {
    required uint32 Count = 1;
}
